src/rbs_location.c

```diff
@@ -15,30 +15,17 @@ void rbs_loc_alloc_children(rbs_allocator_t *allocator, rbs_location_t *loc, int
   loc->children->cap = capacity;
 }
 
-void rbs_loc_add_optional_child(rbs_allocator_t *allocator, rbs_location_t *loc, rbs_constant_id_t name, range r) {
-  // If the location has no children, allocate memory for one child.
-  if (loc->children == NULL) {
-    rbs_loc_alloc_children(allocator, loc, 1);
-  } else if (loc->children->len == loc->children->cap) {
-    // If the location has no more capacity, double the capacity.
-    size_t old_size = RBS_LOC_CHILDREN_SIZE(loc->children->cap);
-    size_t new_size = old_size * 2;
-
-    // Reallocate the children array to the new size.
-    rbs_loc_children *new_children = rbs_allocator_realloc(allocator, loc->children, old_size, new_size, rbs_loc_children);
-    new_children->cap = new_size;
-    loc->children = new_children;
-  }
-
-  // Add a new child to the location.
+void rbs_loc_add_optional_child(rbs_location_t *loc, rbs_constant_id_t name, range r) {
+  assert(loc->children != NULL && "All children should have been pre-allocated with rbs_loc_alloc_children()");
+  assert((loc->children->len + 1 <= loc->children->cap) && "Not enough space was pre-allocated for the children.");
```

Comment:
Isn't this going to be a problem when called from the Ruby API? https://github.com/Shopify/rbs/blob/master/ext/rbs_extension/location.c#L199

Reply:
If I understand the best practice correctly, the functions under `src` should NOT be internal as they're actually public interface to tools like Sorbet.
I think we should, instead, remove the `rbs_` prefix from the functions under `rbs_extension` as they're in fact internal to the Ruby API definition only.
WDYT?

Reply:
Actually, I don't think the naming conflict is much of a problem in `c-api` branch as most of the `rbs_extension` counterparts are removed. For example, there is only one `rbs_loc_add_optional_child` function in that branch.

It's unfortunate that GH search doesn't allow us to search on branch, but I don't feel there's an immediate need for renaming these methods.